test/rpc_api_mock.go

```diff
@@ -52,17 +53,21 @@ func (mock *mockService) Unpin(ctx context.Context, in api.PinSerial, out *struc
 	return nil
 }
 
-func (mock *mockService) PinPath(ctx context.Context, in api.PinSerialWithPath, out *api.CidSerial) error {
-	*out = api.CidSerial{
-		CidTarget: TestCid5,
+func (mock *mockService) PinPath(ctx context.Context, in api.PinOptionsWithPath, out *api.PinSerial) error {
+	_, err := gopath.ParsePath(in.Path)
+	if err != nil {
+		return err
 	}
+	*out = api.PinCid(MustDecodeCid(TestCid5)).ToSerial()
```

Comment:
You may want to add a `TestCidResolved` specifically used as result of resolved things.